src/puppetlabs/dujour/version_check.clj

```diff
@@ -118,7 +128,21 @@
   ([request-values update-server-url callback-fn]
     (validate-config! request-values update-server-url)
     (future
-      (let [server-response (try
+      (let [certname (:certname request-values)
+            request-values (if-not (nil? certname)
+                             (->
+                               request-values
+                               (assoc :host-id (get-hash certname))
+                               (dissoc :certname))
+                             request-values)
+            cacert (:cacert request-values)
+            request-values (if-not (nil? cacert)
```

Comment:
rebinding a name is generally frowned on. You should let `request-values` always refer to what was passed in and name your changed version something else. That's easier if you transform the map all at once. Here's an example using a kitchensink utility function:
```clojure
hashed-values (-> (ks/mapvals get-hash request-values)
                  (set/rename-keys {:certname :host-id
                                    :cacert :site-id}))

```